### PR TITLE
asynq task: clear dependencies between each yield

### PR DESCRIPTION
**Pull request by emorphis**

### CODE REVIEW

File: asynq/_debug.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def __init__(self):
         self.STACK_DUMP_LIMIT              = 0     # In frames, None means infinity
 
         self.ENABLE_COMPLEX_ASSERTIONS = True
+        self.KEEP_DEPENDENCIES         = False  # don't clear dependencies between yields
```
Comment (JelleZijlstra) [inline asynq/_debug.py]: Why do we need this option? When would we set it to true?

Comment (emorphis) [reply asynq/_debug.py]: thought ppl might want it for debugging/profiling and plus it was the easiest way to keep asynq/tests/test_profiler.py working without making big changes

Comment (JelleZijlstra) [reply asynq/_debug.py]: What is the use case for debugging? This seems like a purely internal optimization.

If it's too much trouble to optimize test_profiler.py though (why?), I'm OK with keeping this around.

Comment (emorphis) [reply asynq/_debug.py]: I couldn't figure out a great way to fix test_profiler.py -- the problem is that the profiler output doesn't have much that we can use to reference tasks, so referencing them by their deps is the most direct way we have available.

**Pull request merged**